src/utils/filter.ts

```diff
@@ -43,34 +43,65 @@ export class FilterUtility {
         return false;
       }
 
-      if (typeof filterValue === 'object') {
-        if (Array.isArray(filterValue)) {
-          // if `filterValue` is an array, it is a OneOfFilter
-          // Support OR matches by querying for each values separately,
-          if (!this.matchOneOf(filterValue, indexValue)) {
-            return false;
-          }
-          missingPropertyMatches.delete(filterProperty);
-          continue;
-        } else {
-          // `filterValue` is a `RangeFilter`
-          // range filters cannot range over booleans
-          if (!this.matchRange(filterValue, indexValue as RangeValue)) {
-            return false;
-          }
-          missingPropertyMatches.delete(filterProperty);
-          continue;
+      const matched = Array.isArray(indexValue) ?
+        this.matchArrayFilterIndex(filterValue, indexValue) :
+        this.matchFilterIndex(filterValue, indexValue);
+
+      if (!matched) {
+        return false;
+      }
+
+      missingPropertyMatches.delete(filterProperty);
+    }
+
+    return missingPropertyMatches.size === 0;
```

Comment:
Doesn't look there is a point to keeping `missingPropertyMatches` state, looks like can just return `true`?